Maple2.Server.Game/PacketHandlers/MapleopolyHandler.cs

```diff
@@ -0,0 +1,191 @@
+
+        int currentTilePosition = totalTileValue % blueMarble.Tiles.Count;
+
+        BlueMarbleTile tile = blueMarble.Tiles[currentTilePosition];
+
+        switch (tile.Type) {
+            case BlueMarbleTileType.Item:
+            case BlueMarbleTileType.TreasureTrove:
+                if (!ItemMetadata.TryGet(tile.Item.ItemId, out ItemMetadata? itemMetadata)) {
+                    // TODO: Error
+                    break;
+                }
+                var item = new Item(itemMetadata, tile.Item.ItemRarity, tile.Item.ItemAmount);
+                if (!session.Item.Inventory.Add(item, true)) {
+                    // TODO: Mail to user
+                }
+                break;
+            case BlueMarbleTileType.Backtrack:
+                totalTileValue -= tile.MoveAmount;
+                break;
+            case BlueMarbleTileType.MoveForward:
+                totalTileValue += tile.MoveAmount;
+                break;
+            case BlueMarbleTileType.RoundTrip:
+                totalTileValue += blueMarble.Tiles.Count;
+                break;
+            case BlueMarbleTileType.GoToStart:
+                totalTileValue += blueMarble.Tiles.Count - currentTilePosition;
+                break;
+            case BlueMarbleTileType.Start:
+                // Do nothing
+                break;
+            case BlueMarbleTileType.Lose:
+            case BlueMarbleTileType.RollAgain:
+            case BlueMarbleTileType.Trap:
+            default:
+                Logger.Warning($"Unhandled Mapleopoly tile type: {tile.Type}");
+                break;
+        }
+
+        GameEventUserValue totalTrips = session.GameEventUserValue.Get(GameEventUserValueType.MapleopolyTotalTrips, gameEvent);
+        int.TryParse(totalTrips.Value, out int trips);
+        int totalNewTrips = totalTileValue / blueMarble.Tiles.Count;
+
+        if (totalNewTrips > trips) {
+            int difference = totalNewTrips - trips;
+
+            for (int tripCount = 0; tripCount < difference; tripCount++) {
+                trips++;
+
+                // Check if there's any item to give for every 1 trip
+                var entry1 = blueMarble.Entries.FirstOrDefault(entry => entry.TripAmount == 0);
+                if (entry1 != default && ItemMetadata.TryGet(entry1.Item.ItemId, out ItemMetadata? trip0Metadata)) {
+                    var trip0Item = new Item(trip0Metadata, entry1.Item.ItemRarity, entry1.Item.ItemAmount);
+                    if (!session.Item.Inventory.Add(trip0Item, true)) {
+                        // TODO: mail to player
+                    }
+                }
+
+                // Check if there's any other item to give for hitting a specific number of trips
+                var entry2 = blueMarble.Entries.FirstOrDefault(entry => entry.TripAmount == trips);
+                if (entry2 == default || !ItemMetadata.TryGet(entry2.Item.ItemId, out ItemMetadata? tripMetadata)) {
+                    continue;
```

Comment:
i think it's more clear to follow the same logic pattern as above and not use `continue` here.